### PR TITLE
Modernize

### PR DESCRIPTION
**Pull request by IvanKiral**
### Motivation

Which issue does this fix? Fixes #`issue number`

If no issue exists, what is the fix or new feature? Were there any reasons to fix/implement things that are not obvious?

### Checklist

- [ ] Code follows coding conventions held in this repo
- [ ] Automated tests have been added
- [ ] Tests are passing
- [ ] Docs have been updated (if applicable)
- [ ] Temporary settings (e.g. variables used during development and testing) have been reverted to defaults

### How to test

If manual testing is required, what are the steps?

### CODE REVIEW

**Review by JiriLojda — CHANGES_REQUESTED** [review]
- The `.npmrc` file should hopefully not be necessary now. 
- I would prefer to have all the output (commonjs, es, types...) in a common directory to be able to see it all in one place.
- I would prefer a different name for test files (currently it is `*.spec.ts`). In my opinion something like `*.test.ts` is better.
- Remove the experiment team from the codeowners file and replace them with me or you (if you'd like to be the primary maintainer of this or not).
- Update the tests section in readme. It still mentions loki and storybook.

File: package.json [hunk 1]
```diff
@@ -23,7 +23,7 @@
     "build:commonjs:watch": "npm run build:commonjs -- -w",
     "build:es": "tsc -m es2015 --outDir es --p tsconfig.json",
     "build:es:watch": "npm run build:es -- -w",
-    "build:umd": "npm run webpack:dev && npm run webpack:prod",
+    "build:umd": "vite build",
     "build:umd:watch": "concurrently \"npm run webpack:dev:watch\" \"npm run webpack:prod:watch\"",
     "bundle-stats": "npm run build:umd && webpack-bundle-analyzer dist/stats.json",
     "ci": "npm ci --legacy-peer-deps",
```
[the remark this answers is not in the document]
Comment (IvanKiral) [reply package.json:29]: it is highlightin me build:umd? I thought we wanted to keep umd

File: playwright/index.html [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap@4.5.3/dist/css/bootstrap.min.css" integrity="sha384-TX8t27EcRE3e/ihU7zmQxVncDAy5uIKz4rEkgIXeMed4M0jlfIDPvg6uqKI2xXr2" crossorigin="anonymous">
+        <!-- jQuery and JS bundle w/ Popper.js -->
+    <script src="https://code.jquery.com/jquery-3.5.1.slim.min.js" integrity="sha384-DfXdz2htPH0lsSSs5nCTpuj/zy4C+OGpamoFVy38MVBnE+IbbVYUew+OrCXaRkfj" crossorigin="anonymous"></script>
+    <script src="https://cdn.jsdelivr.net/npm/bootstrap@4.5.3/dist/js/bootstrap.bundle.min.js" integrity="sha384-ho+j7jyWK8fNQe+A12Hb8AhRq26LrZ/JpcUGGOn+Y7RsweNrtN/tE3MoK7ZeZDyx" crossorigin="anonymous"></script>
```
[the remark this answers is not in the document]
Comment (IvanKiral) [reply playwright/index.html:9]: I took it from the original index.html for storybook. Not sure about jQuery, I just think it is a dependency of bootstrap. Bootstrap is there for some little framework for styling.

File: test-components/components/edge-cases/SeparateNodeAddition.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+      element.type = 'button';
+      element.classList.add('btn');
+      element.classList.add('btn-info');
+      element.setAttribute('data-kontent-project-id', 'p');
+      element.setAttribute('data-kontent-language-codename', 'l');
+      element.setAttribute('data-kontent-item-id', 'i');
+      element.setAttribute('data-kontent-element-codename', 'e');
+      element.innerText = 'Button with smart link';
+
+      container.appendChild(element);
```
[the remark this answers is not in the document]
Comment (IvanKiral) [reply test-components/components/edge-cases/SeparateNodeAddition.tsx:17]: lol I must have missed it in diff from AI. Cursor betrayed me :/

File: package.json [hunk 4]
```diff
@@ -65,44 +57,30 @@
...
-    "karma-typescript": "5.5.3",
-    "karma-typescript-es6-transform": "5.5.3",
-    "loki": "0.34.0",
+    "playwright": "^1.50.1",
     "prettier": "2.7.1",
+    "react": "^19.0.0",
+    "react-dom": "^19.0.0",
     "rimraf": "3.0.2",
+    "rollup-plugin-visualizer": "^5.14.0",
     "ts-loader": "9.3.1",
```
[the remark this answers is not in the document]
Comment (IvanKiral) [reply package.json:78]: This creates a nice graph, that analyzes the bundle size.. Right now I don't know how to utilize it, but I managed we could keep it, if we want to look at it in the future... However I updated a config in vite, so it produces a HTML :)
image

File: src/index.ts [hunk 5]
```diff
@@ -1,4 +1,4 @@
-export { default, KontentSmartLinkEvent } from './sdk';
+export { default as KontentSmartLink, KontentSmartLinkEvent } from './sdk';
```
[the remark this answers is not in the document]
Comment (IvanKiral) [reply src/index.ts:1]: You are right; thanks for pointing it out. I tried to npm link to use this package on the next sample app, and it had a problem. However, if I keep the default, then when using the umd, users would have to use KontentSmartLink.default :D (I don't have a problem with that, I guess.)

Maybe we should also release a preview version and give it to Risa to test it on the learn?

Maybe this change of bundler will cause a breaking change?

File: package.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (JiriLojda) [reply package.json:72]: I meant the ts-loader

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (JiriLojda) [reply package.json]: It's for the ci script

File: playwright/index.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (JiriLojda) [reply playwright/index.html:9]: Ok, just leave a comment there explaining that it's just for the test screens.

**Review by JiriLojda — APPROVED** [review]
